### ISSUE TITLE
Update 6.5.0: Enforced Setup Wizard Does not Go Beyond Step 3 example.chat/setup-wizard/3

### ISSUE THREAD

**Issue by chloesoe** [RocketChat/Rocket.Chat#31152]
### Description:

After update from 6.4.8 a new setup wizard is enforced after login (probably related to https://github.com/RocketChat/Rocket.Chat/issues/31149).

After doing all steps and register, the setup wizard always opens at step 3 example.chat/setup-wizard/3 instead of redirect to the main page.

### Steps to reproduce:

1. Use a 6.4.8 installation
2. Update to 6.5.0
3. Login with an admin user
4. Register the organisation
5. Enter mail address in example.chat/setup-wizard/3 and confirm email addres.
6. Checkout https://cloud.rocket.chat/workspaces/, looks like the workspace was added.
7. Go to your chat URL example.chat
8. example.chat/setup-wizard/3  is opened again

### Expected behavior:

Actually no registration at all would be expected, because in the release notes https://github.com/RocketChat/Rocket.Chat/releases/tag/6.5.0 only minor changes were announced. 
Setup wizard is there as minor change, but registration was not mentioned. 
Nevertheless, if needed, I'd expect that after the registration the chat will work and I could open e.g. https://expamle.chat/admin/info 

### Actual behavior:

Connect to all chat URLs will end in step 3 of the wizard:
![image](https://github.com/RocketChat/Rocket.Chat/assets/24636407/4880e0dd-20d8-490d-b081-443eda11ebe9)

### Server Setup Information:

- Version of Rocket.Chat Server: 6.5.0
- Operating System: docker 
- Deployment Method:  docker
- Number of Running Instances: 1 
- DB Replicaset Oplog: 6.0.8 / unknown (oplog Enabled)
- NodeJS Version: unknown
- MongoDB Version: 6.0.8 / unknown (oplog Enabled)

### Client Setup Information

- Desktop App or Browser Version: Brave and Firefox
- Operating System: Linux

### Additional context

n/a

### Relevant logs:

n/a

**Comment by caskelly** [comment]
Thanks for reporting this @chloesoe - We're not seeing this issue elsewhere.  Can you advise on whether you've applied any customizations to this file?

apps/meteor/app/cloud/server/functions/getWorkspaceAccessTokenWithScope.tsapps/meteor/app/cloud/server/functions/getWorkspaceAccessTokenWithScope.ts

@tiagoevanp for visibility

**Comment by tiagoevanp** [comment]
Thanks @caskelly ! @chloesoe , can you provide us more information about the problem like any server log?

**Comment by chloesoe** [comment]
Thanks for your answers. I tried to reproduce it and I think I have found where it happened.

If I open the verification link in the same window as the chat I am registering, then the registration does not complete successfully and I am redirected to step 3. If this happen, I could go to https://cloud.rocket.chat/home and click on "activate trial", then the chat opens properly.
In the logs was nothing really logged. Only once there was  there was a time out:
```
{"level":50,"time":"2023-12-05T06:55:26.744Z","pid":1,"hostname":"rocketchat-666b9dc69f-npqqp","name":"System","msg":"Failed to get confirmation poll from Rocket.Chat Cloud","url":"/api/v2/register/workspace/poll","err":{"type":"Error","message":"Failed to retrieve registration confirmation poll data: Too Many Requests","stack":"Error: Failed to retrieve registration confirmation poll data: Too Many Requests\n    at app/cloud/server/functions/getConfirmationPoll.ts:19:10\n    at /app/bundle/programs/server/npm/node_modules/meteor/promise/node_modules/meteor-promise/fiber_pool.js:43:40"}}

```

And the unregistering is logged properly:
```
{"level":50,"time":"2023-12-05T07:51:51.080Z","pid":1,"hostname":"rocketchat-666b9dc69f-7872w","name":"System","msg":"Server has been unregistered from cloud"}
```


*tl;dr:*
Verification link has to be opened in different browser tab or "activate trial" has to be clicked on https://cloud.rocket.chat/home

**Comment by computan** [comment]
I am having the same issue. It worked few months back but now it's not. I also tried opening verification link to open in different tab but didn't work. I also didn't find activate trial option. My workspace did get registered though, I checked the link: https://cloud.rocket.chat/home to verify that.

**Comment by MaisumAbbas** [comment]
I am having the same issue. It worked few months back but now it's not. I also tried opening verification link to open in different tab but didn't work. I also didn't find activate trial option. My workspace did get registered though, I checked the link: https://cloud.rocket.chat/home to verify that.

**Comment by sforsythe-sv** [comment]
> I am having the same issue. It worked few months back but now it's not. I also tried opening verification link to open in different tab but didn't work. I also didn't find activate trial option. My workspace did get registered though, I checked the link: https://cloud.rocket.chat/home to verify that.

Hi @MaisumAbbas I was also having this issue and none of the previous suggestions worked. What I did, (because I am using a reverse proxy to public access and register rocketchat) is - replaced the ROOT_URL value with the URL from the proxy domain so instead of http://192.xxx.xxx.xx it was now https://mydomain.com - I hope this helps!

**Comment by sebastianleecn** [comment]
same issue, i'll try the way, thanks

> > I am having the same issue. It worked few months back but now it's not. I also tried opening verification link to open in different tab but didn't work. I also didn't find activate trial option. My workspace did get registered though, I checked the link: https://cloud.rocket.chat/home to verify that.
> 
> Hi [@MaisumAbbas](https://github.com/MaisumAbbas) I was also having this issue and none of the previous suggestions worked. What I did, (because I am using a reverse proxy to public access and register rocketchat) is - replaced the ROOT_URL value with the URL from the proxy domain so instead of http://192.xxx.xxx.xx it was now https://mydomain.com - I hope this helps!

same issue, i'll try the way, thanks
